File: identhub/src/main/java/de/solarisbank/identhub/domain/verification/phone/ConfirmVerificationPhoneUseCase.java
package de.solarisbank.identhub.domain.verification.phone;

import org.jetbrains.annotations.NotNull;

import de.solarisbank.identhub.data.TransactionAuthenticationNumber;
import de.solarisbank.identhub.data.verification.phone.model.VerificationPhoneResponse;
import de.solarisbank.identhub.domain.usecase.SingleUseCase;
import io.reactivex.Single;
import io.reactivex.android.schedulers.AndroidSchedulers;

public class ConfirmVerificationPhoneUseCase extends SingleUseCase<String, VerificationPhoneResponse> {
    private final VerificationPhoneRepository verificationPhoneRepository;

    public ConfirmVerificationPhoneUseCase(VerificationPhoneRepository verificationPhoneRepository) {
        this.verificationPhoneRepository = verificationPhoneRepository;
    }

    @NotNull
    @Override
    protected Single<VerificationPhoneResponse> invoke(String confirmToken) {
        return verificationPhoneRepository.confirmToken(new TransactionAuthenticationNumber(confirmToken))
                .observeOn(AndroidSchedulers.mainThread());
    }
}

File: identhub/src/main/java/de/solarisbank/identhub/domain/verification/phone/ConfirmVerificationPhoneUseCase.kt
package de.solarisbank.identhub.domain.verification.phone

import de.solarisbank.identhub.session.data.TransactionAuthenticationNumber
import de.solarisbank.identhub.session.data.verification.phone.model.VerificationPhoneResponse
import de.solarisbank.sdk.data.entity.NavigationalResult
import de.solarisbank.sdk.domain.usecase.SingleUseCase
import io.reactivex.Single
import io.reactivex.android.schedulers.AndroidSchedulers

class ConfirmVerificationPhoneUseCase(private val verificationPhoneRepository: VerificationPhoneRepository) : SingleUseCase<String, VerificationPhoneResponse>() {
    override fun invoke(confirmToken: String): Single<NavigationalResult<VerificationPhoneResponse>> {
        return verificationPhoneRepository.confirmToken(TransactionAuthenticationNumber(confirmToken))
                .map { NavigationalResult(it) }
                .observeOn(AndroidSchedulers.mainThread())
    }
}
